<!DOCTYPE html>
<html lang="en-US">
<!--********************************************-->
<!--*       Generated from PreTeXt source      *-->
<!--*       on 2022-07-19T13:33:02-04:00       *-->
<!--*   A recent stable commit (2020-08-09):   *-->
<!--* 98f21740783f166a773df4dc83cab5293ab63a4a *-->
<!--*                                          *-->
<!--*         https://pretextbook.org          *-->
<!--*                                          *-->
<!--********************************************-->
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="robots" content="noindex, nofollow">
</head>
<body>
<article class="li"><h6 xmlns:svg="http://www.w3.org/2000/svg" class="heading">
<span class="type">Item</span><span class="space"> </span><span class="codenumber">3</span><span class="period">.</span>
</h6>Bioinformaticists are often interested in frequent substrings within a genome that occur with high frequency in a particular region. Update your function to count the frequency of ATGATCAAG within a sliding window of 300 nucleotides along the Vibrio cholerae genome. Your function should return two items - the maximum number of times ATGATCAAG occurs within the Vibrio cholerae genome inside of the 300-nucleotide window and the starting index of the window. Be sure to clearly indicate whether you are using indexing beginning from 0 or 1. What is the run-time complexity of this algorithm?</article><span class="incontext"><a href="sec_p2-alg-complexity-DNA-replication.html#li-15">in-context</a></span>
</body>
</html>
